sync/sync.go

```diff
@@ -42,13 +45,17 @@ type Reader interface {
 	StartingBlockNumber() (uint64, error)
 	HighestBlockHeader() *core.Header
 	SubscribeNewHeads() HeaderSubscription
+
+	Pending() (*Pending, error)
+	PendingBlock() *core.Block
+	PendingState() (core.StateReader, func() error, error)
 }
 
 // This is temporary and will be removed once the p2p synchronizer implements this interface.
 type NoopSynchronizer struct{}
 
 func (n *NoopSynchronizer) StartingBlockNumber() (uint64, error) {
-	return 0, nil
+	return 0, errors.New("StartingBlockNumber not implemented")
```

Comment:
nit: to make it consistent maybe we should pass this message "StartingBlockNumber() not implemented"

Reply:
Done